### PR TITLE
Append file name to test asset

### PR DESCRIPTION
**Pull request by wli3**
So that tests with the same method name would not conflict

### CODE REVIEW

**Review by danmoseley — APPROVED** [review]
This should work.

It assumes that there aren't any cases where two test classes (that conflict) are in the same file.

**Comment by wli3** [comment]
@dotnet/dotnet-cli please review

File: src/Tests/Microsoft.NET.TestFramework/TestAssetsManager.cs [hunk 1]
```diff
@@ -36,13 +36,15 @@ public TestAssetsManager(ITestOutputHelper log)
         public TestAsset CopyTestAsset(
             string testProjectName,
             [CallerMemberName] string callingMethod = "",
+            [CallerFilePath] string callerFilePath = null,
```
Comment (akoeplinger) [inline src/Tests/Microsoft.NET.TestFramework/TestAssetsManager.cs:39]: we should probably make the same change for `CreateTestProject()`

File: src/Tests/Microsoft.NET.TestFramework/TestAssetsManager.cs [hunk 2]
```diff
@@ -109,13 +111,13 @@ public string GetAndValidateTestProjectDirectory(string testProjectName, string
 
         public static string GetTestDestinationDirectoryPath(
             string testProjectName,
-            string callingMethod,
+            string callingMethodAndFileName,
             string identifier)
```
Comment (akoeplinger) [inline src/Tests/Microsoft.NET.TestFramework/TestAssetsManager.cs:115]: I think we could simplify this by just having a single `identifier` that combines `testProjectName`, `callingMethodAndFileName` and `identifier`. With the current setup the directory name will always be longer than 24 chars anyway so we'll always do the hashing.

That'd allow us to pass in the full path to the file name which would address @danmoseley's point about having the same file name with same test names in different directories.

Comment (wli3) [reply src/Tests/Microsoft.NET.TestFramework/TestAssetsManager.cs:115]: that is pretty much what I am doing. Note, on mac and linux, we do not hash them, so we don't want the folder name to be super long (it is hard for dev to debug).

**Pull request merged**